Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/revocation/update/RevocationListUpdater.kt

```diff
@@ -0,0 +1,74 @@
+package de.rki.coronawarnapp.covidcertificate.revocation.update
+
+import androidx.annotation.VisibleForTesting
+import de.rki.coronawarnapp.covidcertificate.common.certificate.CertificateProvider
+import de.rki.coronawarnapp.covidcertificate.revocation.storage.RevocationRepository
+import de.rki.coronawarnapp.tag
+import de.rki.coronawarnapp.util.TimeAndDateExtensions.toLocalDateUtc
+import de.rki.coronawarnapp.util.TimeStamper
+import de.rki.coronawarnapp.util.coroutine.AppScope
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.flow.drop
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+import org.joda.time.Instant
+import timber.log.Timber
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class RevocationListUpdater @Inject constructor(
+    @AppScope private val appScope: CoroutineScope,
+    private val timeStamper: TimeStamper,
+    private val certificatesProvider: CertificateProvider,
+    private val revocationUpdateSettings: RevocationUpdateSettings,
+    private val revocationRepository: RevocationRepository,
+) {
+
+    private val mutex = Mutex()
+
+    init {
+        appScope.launch {
```

Comment:
the forced update shall only be triggered when a new cert gets registered.
Is it possible to use distingUntilChanged {} with some little custom logic to compare the size of the emission?

Reply:
This flow is a state flow, so I will quote your words. it has it be design 😉

Reply:
Yes, stateful flow is distintUntilChanged by design but it relies on equalsTo for that. This would mean the update gets triggered when the count reduces, like when a cert gets deleted by the user or after 30 days in recycle bin. So my thought was to use something like distinctUntilChanged { newEmission > oldEmission }. Would that work?